### ISSUE TITLE
ShellCheck fails to run using pre-commit (Debian-based distro/shell: bash)

### ISSUE THREAD

**Issue by bowmanjc** [koalaman/shellcheck#2646]
#### For bugs
- Rule Id (if any, e.g. SC1000): n/a
- My shellcheck version (`shellcheck --version` or "online"): 0.9.0
- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [X] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

# .pre-commit-config.yaml
repos:
-   repo: https://github.com/koalaman/shellcheck-precommit
    rev: v0.9.0
    hooks:
    -   id: shellcheck
#       args: ["--severity=warning"]  # Optionally only show errors and warnings
```

#### Here's what shellcheck currently says:
```sh
$ pre-commit run                                                18:00:00 Mon 26
ShellCheck v0.9.0........................................................Failed
- hook id: shellcheck
- exit code: 1

Executable `docker` not found
```

**Comment by brother** [comment]
> Executable `docker` not found

you are sure that your docker setup is indeed correct and the binary is in your PATH?

**Comment by bowmanjc** [comment]
Sorry, that's where I wasn't very clear. I do not have docker installed. Shellcheck doesn't need docker to run. Why would shellcheck need docker to run using pre-commit, when pre-commit doesn't require docker either?

**Comment by brother** [comment]
Maybe this is the first extension using docker in your case. I haven’t used pre-commit myself but the repo look to include docker references.  
see https://github.com/koalaman/shellcheck-precommit/blob/master/.pre-commit-hooks.yaml

**Comment by bowmanjc** [comment]
I think that's where the problem is. There shouldn't be any reason that shellcheck is dependent on docker to run under pre-commit. Neither shellcheck, nor pre-commit, require docker individually, so the combination shouldn't require docker either.

**Comment by austin987** [comment]
> I think that's where the problem is. There shouldn't be any reason that shellcheck is dependent on docker to run under pre-commit. Neither shellcheck, nor pre-commit, require docker individually, so the combination shouldn't require docker either.

Shellcheck itself doesn't depend on docker. But if you want to run a docker container on pre-commit, docker needs to be installed. That's not a shellcheck issue.

Alternatively, you can download the binary and run that directly. Or checkout shellcheck and build it from source. Or ask whoever maintains pre-commit to include docker in their image.

**Comment by koalaman** [comment]
pre-commit hooks are expected to be cross-platform. Here's a list of [languages/runners](https://pre-commit.com/#new-hooks) support by pre-commit, which includes things like `python`,  `lua`, and `node`, and `docker_image`. The latter is really the only one that can run a Haskell program in a portable way. pre-commit implements this by assuming you already have Docker installed, which is an understandable concession.

You can have a look at this third-party hook instead, which even predates the official one: https://github.com/shellcheck-py/shellcheck-py  . It uses the Python runner with a wrapper that downloads an appropriate binary for the current platform.
